climakitae/explore/warming.py

```diff
@@ -218,27 +236,36 @@ def get_sliced_data(y, level, years, window=15, anom="Yes"):
 
         if anom == "Yes":
             # Find the anomaly
-            anom_val = y.sel(time=slice("1981", "2010")).mean("time")
+            anom_val = y.sel(time=slice("1980", "2010")).mean("time")
             sliced = y.sel(time=slice(str(start_year), str(end_year))) - anom_val
         else:
             # Finding window slice of data
             sliced = y.sel(time=slice(str(start_year), str(end_year)))
 
-        # Resetting time index for each data array so they can overlap and save storage space
-        sliced["time"] = sliced.time - sliced.time[0]
+        # Resetting and renaming time index for each data array so they can overlap and save storage space
+        sliced["time"] = np.arange(-len(sliced.time) / 2, len(sliced.time) / 2)
 
         # Assigning `centered_year` as a coordinate to the DataArray
         sliced = sliced.assign_coords({"centered_year": centered_year})
 
         return sliced
 
     else:
+
+        # This creates an approximately appropriately sized DataArray to be dropped later
+        if y.frequency == "monthly":
+            time_freq = 12
+        elif y.frequency == "daily":
+            time_freq = 365
+        elif y.frequency == "hourly":
+            time_freq = 8760
+
         y = y.isel(
-            time=slice(0, window * 2 * 365)
+            time=slice(0, window * 2 * time_freq)
         )  # This is to create a dummy slice that conforms with other data structure. Can be re-written to something more elegant.
 
         # Creating attributes
-        y["time"] = y.time - y.time[0]
+        y["time"] = np.arange(-len(y.time) / 2, len(y.time) / 2)
```

Comment:
Should this be 

```
np.concatenate(
                [
                    np.arange(-len(metric_data.time) / 2, 0),
                    np.arange(1, len(metric_data.time) / 2 + 1),
                ]
            )
```

instead?

Reply:
The current solution works, even though centered_year is not at the exact center.. In fact the output would be easier to work with as the array is continuous. Good thinking about time from center though (and renaming the dimension)!